packages/twenty-server/src/database/commands/upgrade-version/0-41/0-41-migrate-relations-to-field-metadata.command.ts

```diff
@@ -0,0 +1,137 @@
+    options: ActiveWorkspacesCommandOptions,
+    workspaceIds: string[],
+  ): Promise<void> {
+    this.logger.log('Running command to create many to one relations');
+
+    if (isCommandLogger(this.logger)) {
+      this.logger.setVerbose(options.verbose ?? false);
+    }
+
+    const workspacePromises = workspaceIds.map((workspaceId, index) =>
+      this.processWorkspace(workspaceId, index, workspaceIds.length),
+    );
+
+    try {
+      await Promise.all(workspacePromises);
+
+      this.logger.log(chalk.green('Command completed!'));
+    } catch (error) {
+      this.logger.log(chalk.red('Error in workspace'));
+    }
+  }
+
+  private async processWorkspace(
+    workspaceId: string,
+    index: number,
+    total: number,
+  ): Promise<void> {
+    try {
+      this.logger.log(
+        `Running command for workspace ${workspaceId} ${index + 1}/${total}`,
+      );
+
+      const fieldMetadataCollection = (await this.fieldMetadataRepository.find({
+        where: { workspaceId, type: FieldMetadataType.RELATION },
+        relations: ['fromRelationMetadata', 'toRelationMetadata'],
+      })) as unknown as FieldMetadataEntity<FieldMetadataType.RELATION>[];
+
+      if (!fieldMetadataCollection.length) {
+        this.logger.log(
+          chalk.yellow(
+            `No relation field metadata found for workspace ${workspaceId}.`,
+          ),
+        );
+
+        return;
+      }
+
+      const fieldMetadataToUpdateCollection = fieldMetadataCollection.map(
+        (fieldMetadata) => this.mapFieldMetadata(fieldMetadata),
+      );
+
+      if (fieldMetadataToUpdateCollection.length > 0) {
+        await this.fieldMetadataRepository.save(
+          fieldMetadataToUpdateCollection,
+        );
+      }
+
+      this.logger.log(
+        chalk.green(`Command completed for workspace ${workspaceId}.`),
+      );
+    } catch {
+      this.logger.log(chalk.red(`Error in workspace ${workspaceId}.`));
+    }
+  }
+
+  private mapFieldMetadata(
+    fieldMetadata: FieldMetadataEntity<FieldMetadataType.RELATION>,
+  ): FieldMetadataEntity<FieldMetadataType.RELATION> {
+    const relationMetadata =
+      fieldMetadata.fromRelationMetadata ?? fieldMetadata.toRelationMetadata;
+
+    const relationDirection = deduceRelationDirection(
+      fieldMetadata,
+      relationMetadata,
+    );
+    let relationType = relationMetadata.relationType as unknown as RelationType;
+
+    if (
+      relationDirection === RelationDirection.TO &&
+      relationType === RelationType.ONE_TO_MANY
+    ) {
+      relationType = RelationType.MANY_TO_ONE;
+    }
+
+    const targetFieldMetadataId =
+      relationDirection === RelationDirection.FROM
+        ? relationMetadata.toFieldMetadataId
+        : relationMetadata.fromFieldMetadataId;
+
+    return {
+      ...fieldMetadata,
+      settings: {
+        ...fieldMetadata.settings,
+        relationType,
+        onDelete: relationMetadata.onDeleteAction,
+      },
+      targetFieldMetadataId,
```

Comment:
shouldn't we also add the targetObjectFieldMetadataId? It's not useful from data consistency perspective but will likely ease the dev xp?

(the case I'm thinking about: I'm generating the graphql API and I'm looping over RELATION field, i need to add the type of the targetObject but I only have the targetFieldMetadataId. To be efficient, I would need to build an additional map of {fieldMetadataId => object}, additional to the map we already have {objectMetadataId => object}